### PR TITLE
Refactor into multiplatform project, target JS

### LINKED ISSUE #151: Cannot build SpotifyClientAPI with expired token
**Description**
If you create an instance with an expired token an exception gets thrown in the init method. This should not be, because you aren't even able to refresh your access token with a valid refresh token. 
This is because the user profile get loaded in the init function, which throws the exception.

**To Reproduce**
Build an instance of the SpotifyClientAPI with an expired token.

**Stack**
``` 
com.adamratzman.spotify.models.BadRequestException: Received Status Code 401. Error cause: Invalid access token
	at com.adamratzman.spotify.http.SpotifyEndpoint.handleResponse(Endpoints.kt:110) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.http.SpotifyEndpoint.execute(Endpoints.kt:70) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.http.SpotifyEndpoint.execute$default(Endpoints.kt:49) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.http.SpotifyEndpoint.get$spotify_api_kotlin(Endpoints.kt:25) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.endpoints.client.ClientUserAPI$getUserProfile$1.get(ClientUserAPI.kt:29) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.endpoints.client.ClientUserAPI$getUserProfile$1.get(ClientUserAPI.kt:16) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.SpotifyRestAction.complete(SpotifyRestAction.kt:34) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.SpotifyClientAPI.<init>(SpotifyAPI.kt:354) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.SpotifyApiBuilderDsl.buildClient(Builder.kt:460) ~[spotify-api-kotlin-2.3.08.jar:na]
	at com.adamratzman.spotify.SpotifyApiBuilderDsl.buildClient(Builder.kt:400) ~[spotify-api-kotlin-2.3.08.jar:na]
```

### LINKED ISSUE #153: Model issues
- PlayHistory context can be null sometimes (Spotify bug, but we should handle)
- PlayHistory context uri can be any of playlist, artist, album
- CursorBasedPagingObject total is sometimes not sent (need to set to default)

See https://github.com/adamint/spotify-web-api-kotlin/compare/hackathon?expand=1

### PR DESCRIPTION
**Pull request by adamint**
3.0.0-rc.2 resolves #151, resolves #153

### CODE REVIEW

File: .idea/artifacts/spotify_web_api_kotlin_js_1_0_SNAPSHOT.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<component name="ArtifactManager">
+  <artifact type="jar" name="spotify-web-api-kotlin-js-1.0-SNAPSHOT">
+    <output-path>$PROJECT_DIR$/build/libs</output-path>
+    <root id="archive" name="spotify-web-api-kotlin-js-1.0-SNAPSHOT.jar">
+      <element id="module-output" name="spotify-web-api-kotlin.spotify-web-api-kotlin.jsMain" />
+    </root>
+  </artifact>
+</component>
```
Comment (molikuner) [inline .idea/artifacts/spotify_web_api_kotlin_js_1_0_SNAPSHOT.xml]: You added this file even though it is ignored in the `.gitignore`. Do you want to have it?

File: src/commonMain/kotlin/com.adamratzman.spotify/Builder.kt [hunk 2]
```diff
@@ -311,21 +313,23 @@ class SpotifyClientApiBuilder(
         }
     }
 
-    override fun buildAsync(consumer: (SpotifyClientAPI) -> Unit) = Thread { consumer(build()) }.start()
+    override suspend fun buildAsync(consumer: (SpotifyClientApi) -> Unit) = coroutineScope {
+        withContext(Dispatchers.Default) { consumer(build()) }
+    }
```
Comment (molikuner) [inline src/commonMain/kotlin/com.adamratzman.spotify/Builder.kt]: As I already said, this doesn't make any sense. This function blocks until the consumer finishes. I wouldn't call that async.

File: src/commonMain/kotlin/com.adamratzman.spotify/SpotifyRestAction.kt [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    /**
+     * Invoke [supplier] and synchronously retrieve [T]
+     */
+    fun complete(): T {
+        hasRunBacking = true
+        return try {
+            supplier().also { hasCompletedBacking = true }
+        } catch (e: Throwable) {
+            throw e
+        }
```
Comment (molikuner) [inline src/commonMain/kotlin/com.adamratzman.spotify/SpotifyRestAction.kt]: Why is this wrapped in a `try/catch`? This doesn't change anything.

File: src/jsMain/kotlin/com/adamratzman/spotify/utils/Concurrency.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+internal actual fun CoroutineScope.schedule(
+    quantity: Int,
+    timeUnit: TimeUnit,
+    consumer: () -> Unit
+) {
+    launch {
+        delay(timeUnit.toMillis(quantity.toLong()))
+    }
+}
```
Comment (molikuner) [inline src/jsMain/kotlin/com/adamratzman/spotify/utils/Concurrency.kt]: This doesn't execute consumer.

File: src/commonMain/kotlin/com.adamratzman.spotify/SpotifyRestAction.kt (hunk 3 above)
Comment (adamint) [reply src/commonMain/kotlin/com.adamratzman.spotify/SpotifyRestAction.kt]: Good question. Thanks for catching these idiotic changes 😄

**Pull request merged**